### PR TITLE
Rename s3 input to aws-s3 input

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to rename Filebeat `s3` input to `aws-s3` input.
Closes https://github.com/elastic/integrations/issues/488

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

1. Test the Filebeat `aws-s3` input:
```
filebeat.inputs:
- type: aws-s3
  enabled: true
  queue_url: https://sqs.eu-central-1.amazonaws.com/428152502467/filebeat-aws-elb-test
  credential_profile_name: elastic-beats
```
This should start `aws-s3` input with no error.

2. Test Filebeat filesets that use `aws-s3` input:
```
./filebeat modules enable aws
```
and change `modules.d/aws.yml` to enable a fileset in `aws` module. For example:
```
- module: aws
  cloudtrail:
    enabled: true
    var.queue_url: https://sqs.us-east-1.amazonaws.com/428152502467/test-fb-ks
    var.credential_profile_name: elastic-beats
```
Then run Filebeat and you should see no error. If there are new logs going into the S3 which generates new SQS messages in `test-fb-ks`, you should also see the events in Elasticsearch.

### CODE REVIEW

**Comment by bmorelli25** [comment]
For the doc build, change this line:

https://github.com/elastic/beats/blame/18fb50493f6cd60eed1661caa3c4939de59db20e/filebeat/docs/filebeat-options.asciidoc#L79

It should be:

```
* <<{beatname_lc}-input-aws-s3>>
```

When I make this change the PR builds locally 🎉

File: filebeat/docs/filebeat-options.asciidoc [hunk 1]
```diff
@@ -63,6 +63,7 @@ subdirectories of a directory.
 You can configure {beatname_uc} to use the following inputs:
 
 * <<{beatname_lc}-input-aws-cloudwatch>>
+* <<{beatname_lc}-input-aws-s3>>
```
Comment (bmorelli25) [inline filebeat/docs/filebeat-options.asciidoc:66]: Nice catch on the alphabetizing!

Comment (kaiyan-sheng) [reply filebeat/docs/filebeat-options.asciidoc:66]: Ha thank you for your help!!

**Review by leehinman — CHANGES_REQUESTED** [review]
I think `filebeat/docs/aws-credentials-examples.asciidoc` and `x-pack/filebeat/_meta/config/filebeat.inputs.reference.xpack.yml.tmpl` need updating too.

otherwise looks great.

**Comment by kaiyan-sheng** [comment]
@leehinman Good catch! It's fixed now, thank you!

**Review by leehinman — APPROVED** [review]

**Pull request merged**